src/gobstuf/stuf/brp/response_mapping.py

```diff
@@ -223,6 +223,30 @@ def related(self):  # pragma: no cover
             'ouders': 'BG:inp.heeftAlsOuders',
         }
 
+    def sort_ouders(self, ouders: list):
+        """Sorts ouders by:
+
+        - geboortedatum descending
+        - geslachtsaanduiding (vrouw, man, onbekend, ..)
+        - geslachtsnaam ascending
+        - voornamen ascending
+
+        Adds ouderAanduiding attribute based on the ordering (ouder1, ouder2)
+
+        :param ouders:
+        :return:
+        """
+        geslachtsaanduiding_order = ['vrouw', 'man', 'onbekend']
+        sorted_ouders = sorted(ouders, key=lambda o: (
+            -(int(str(o.get('geboorte', {}).get('datum', {}).get('datum', 0)).replace('-', ''))),
+            geslachtsaanduiding_order.index(o['geslachtsaanduiding'])
+            if o.get('geslachtsaanduiding') in geslachtsaanduiding_order else 99,
+            o.get('naam', {}).get('geslachtsnaam', 'zzzzzzzzzz').lower(),
+            o.get('naam', {}).get('voornamen', 'zzzzzzzzzz').lower(),
+        ))
```

Comment:
DIt is wel een beetje magic...

Reply:
in lib/communicatie.py is een functie _get_value(). Is deze niet bruikbaar? Moet dan naar utils natuurlijk en misschien een optionele default parameter hebben om het nog wat meer leesbaar te maken.

Ik had er aardig wat tijd voor nodig om te ontdekken hoe deze functie werkte. Sorteren op een tuple is wel een mooie vinding om op multiple keys te sorteren.

Al met al kom ik uit op zoiets als:
`
    def sort_ouders(self, ouders: list):
        """Sorts ouders by:

        - geboortedatum descending
        - geslachtsaanduiding (vrouw, man, onbekend, ..)
        - geslachtsnaam ascending
        - voornamen ascending

        Adds ouderAanduiding attribute based on the ordering (ouder1, ouder2)

        :param ouders:
        :return:
        """
        def ouder_sorter(ouder):
            _get_value = 'extract from communicatie.py and rename....'
            from gobstuf.indications import Geslachtsaanduiding
            
            MAX_NAAM = 'zzzzzzzzzz'  # Value that is expected to compare above any real naam
            geslachtsaanduiding_order = {
                Geslachtsaanduiding.indications.VROUW: 0,
                Geslachtsaanduiding.indications.MAN: 1,
                Geslachtsaanduiding.indications.ONBEKEND: 2,
                None: 3
            }

            # First sort key is geboortedatum descending
            geboorte = (_get_value(ouder, 'geboorte', 'datum', 'datum') or '0000-00-00').replace('-', '')
            geboorte = -(int(geboorte)) # latest first
            
            # Second key is geslachtsaanduiding on geslachtsaanduiding_order
            geslacht = geslachtsaanduiding_order[_get_value(ouder, 'geslachtsaanduiding')]
            
            # Third key is geslachtsnaam ascending
            geslachtsnaam = _get_value('naam', 'geslachtsnaam')
            geslachtsnaam = geslachtsnaam.lower() if geslachtsnaam else MAX_NAAM
            
            # Fourth key is voornamen ascending
            voornamen = _get_value('naam', 'voornamen')
            voornamen = voornamen.lower() if voornamen else MAX_NAAM
            
            return (geboorte, geslacht, geslachtsnaam, voornamen)

        sorted_ouders = sorted(ouders, key=ouder_sorter)

        return [{**ouder, 'ouderAanduiding': f'ouder{idx + 1}'} for idx, ouder in enumerate(sorted_ouders)]

`

Misschien ook niet perfect maar wat beschrijvender